etrago/cluster/snapshot.py

```diff
@@ -68,30 +83,44 @@ def tsam_cluster(timeseries_df, typical_periods=10, how='daily'):
 
     if how == 'daily':
         hours = 24
+        period = ' days'
     if how == 'weekly':
         hours = 168
+        period = ' weeks'
+
+    print('Snapshot clustering to ' + str(typical_periods) + period + 
+          ' using extreme period method: ' + extremePeriodMethod)
 
     aggregation = tsam.TimeSeriesAggregation(
         timeseries_df,
         noTypicalPeriods=typical_periods,
+        extremePeriodMethod = extremePeriodMethod,
+        addPeakMin = ['residual_load'],
+        addPeakMax = ['residual_load'],
         rescaleClusterPeriods=False,
         hoursPerPeriod=hours,
         clusterMethod='hierarchical')
 
+
     timeseries = aggregation.createTypicalPeriods()
     cluster_weights = aggregation.clusterPeriodNoOccur
+    clusterOrder =aggregation.clusterOrder
+    clusterCenterIndices= aggregation.clusterCenterIndices
 
-    # get the medoids/ the clusterCenterIndices
-    clusterCenterIndices = aggregation.clusterCenterIndices
+    if extremePeriodMethod  != 'None':
+        for i in aggregation.extremePeriods.keys():
+            clusterCenterIndices.insert(
+                    aggregation.extremePeriods[i]['newClusterNo'],
+                    aggregation.extremePeriods[i]['stepNo'])
 
     # get all index for every hour of that day of the clusterCenterIndices
-    start = []
+    start=[]
     # get the first hour of the clusterCenterIndices (days start with 0)
     for i in clusterCenterIndices:
-        start.append(i * hours)
+        start.append(i*hours)
 
     # get a list with all hours belonging to the clusterCenterIndices
-    nrhours = []
+    nrhours=[]
```

Comment:
Were these style changes intentional?

Reply:
No, they were not. I can't remember why I changed this, it's not pep8-conform. I can re-change that

Reply:
that would be nice.

Reply:
I did this in my latest commit. Looking at the old commit it only effected three lines, or did you see other issues?